packages/api/src/command/medical/patient/create-patient-settings.ts

```diff
@@ -0,0 +1,172 @@
+  cxId,
+  patientIds,
+  subscribeTo = { adt: false },
+}: PatientSettingsUpsertProps): Promise<PatientSettingsUpsertResults> {
+  const { log } = out(`upsertPatientSettingsForPatientList - cx ${cxId}`);
+
+  const { validPatientIds, invalidPatientIds: patientsNotFound } = await verifyPatients({
+    patientIds,
+    cxId,
+  });
+
+  if (validPatientIds.length === 0) {
+    throw new BadRequestError(`No valid patients found`);
+  }
+
+  const patientsFoundAndUpdated = await upsertPatientSettings({
+    patientIds: validPatientIds,
+    cxId,
+    subscribeTo,
+  });
+
+  log(`Updated settings for ${patientsFoundAndUpdated} patients`);
+  return { patientsFoundAndUpdated, patientsNotFound };
+}
+
+/**
+ * Upserts patient settings for the given customer with specific patient IDs.
+ *
+ * @param cxId The customer ID
+ * @param subscribeTo The subscriptions to enable or disable.
+ * @returns The number of patients updated and the list of patients not found
+ */
+export async function upsertPatientSettingsForCx({
+  cxId,
+  subscribeTo = { adt: false },
+}: PatientSettingsUpsertForCxProps): Promise<PatientSettingsUpsertResults> {
+  const { log } = out(`upsertPatientSettingsForCx - cx ${cxId}`);
+
+  const patientIds = await getPatientIds({ cxId });
+
+  if (!patientIds.length) {
+    log(`No patients found for cx ${cxId}`);
+    return { patientsFoundAndUpdated: 0 };
+  }
+
+  const patientsFoundAndUpdated = await upsertPatientSettings({
+    patientIds,
+    cxId,
+    subscribeTo,
+  });
+
+  log(`Upserted settings for all CX patients. Total: ${patientsFoundAndUpdated}`);
+  return { patientsFoundAndUpdated };
+}
+
+async function verifyPatients({
+  patientIds,
+  cxId,
+}: {
+  patientIds: string[];
+  cxId: string;
+}): Promise<{
+  validPatientIds: string[];
+  invalidPatientIds: string[];
+}> {
+  if (patientIds.length < 1) {
+    return {
+      validPatientIds: [],
+      invalidPatientIds: [],
+    };
+  }
+
+  const patients = await PatientModel.findAll({
+    where: { id: patientIds, cxId },
+    attributes: ["id"],
+  });
+  const foundPatientIds = new Set(patients.map(p => p.id));
+  const invalidPatientIds = patientIds.filter(id => !foundPatientIds.has(id));
+  return {
+    validPatientIds: Array.from(foundPatientIds),
+    invalidPatientIds,
+  };
+}
+
+async function upsertPatientSettings({
+  patientIds,
+  cxId,
+  subscribeTo,
+}: {
+  patientIds: string[];
+  cxId: string;
+  subscribeTo: Subscriptions;
+}): Promise<number> {
+  const existingSettings = await PatientSettingsModel.findAll({
+    where: { patientId: patientIds, cxId },
+  });
+  const existingSettingsMap = new Map(existingSettings.map(s => [s.patientId, s]));
+
+  const upserts = patientIds.map(patientId => ({
+    id: existingSettingsMap.get(patientId)?.id ?? uuidv7(),
+    cxId,
+    patientId,
+    subscribeTo,
+  }));
+
+  await PatientSettingsModel.bulkCreate(upserts, {
+    updateOnDuplicate: ["subscribeTo"],
+  });
```

Comment:
Not sure what the default value is, but we should make sure we're not returning unnecessary data from the DB - we can use `returning: false` for this. And I think it makes it clearer to have the property there, even if the default is false.

See `createOrUpdateCoverageEnhancements()` for ref.